api.py

```diff
@@ -21,14 +21,28 @@ def api_main():
     return jsonify(get_submission_list())
 
 
-@app.route('/<path:path>')
-def server(path):
-    return send_from_directory('public', path)
+@app.route('/ann/<path:path>')
+def ann(path):
+    announcements = get_submission_list()
+    announcement = next((x for x in announcements if x['digest'] == path), None)
```

Comment:
I think it would be faster to just find the announcement by digest from the database. Remember to check whether the announcement is present or not.

Reply:
fix it